### PR TITLE
#8440 Allow for API Port to be specified when using proxmox_kvm

### PR DESCRIPTION
**Pull request by tgouverneur**
##### SUMMARY

Implements #8440 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
modules_utils/proxmox.py

##### ADDITIONAL INFORMATION

N/A -- see #8440

### CODE REVIEW

File: changelogs/fragments/8440-allow-api-port-specification.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - proxmox - allow specification of the API Port when using proxmox_* (https://github.com/ansible-collections/community.general/issues/8440).
```
Comment (felixfontein) [inline changelogs/fragments/8440-allow-api-port-specification.yaml:2]:
```suggestion
  - proxmox - allow specification of the API Port when using proxmox_* (https://github.com/ansible-collections/community.general/issues/8440, https://github.com/ansible-collections/community.general/pull/8441).
```

File: plugins/module_utils/proxmox.py [hunk 2]
```diff
@@ -29,6 +29,9 @@ def proxmox_auth_argument_spec():
                       required=True,
                       fallback=(env_fallback, ['PROXMOX_HOST'])
                       ),
+        api_port=dict(type='str',
```
Comment (felixfontein) [inline plugins/module_utils/proxmox.py:32]: Why string, and not integer?

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Please note that you also have to document the new option in `plugins/doc_fragments/proxmox.py`.

File: plugins/module_utils/proxmox.py (hunk 2 above)
Comment (tgouverneur) [reply plugins/module_utils/proxmox.py:32]: Will change that

**Comment by tgouverneur** [comment]
> Thanks for your contribution! Please note that you also have to document the new option in `plugins/doc_fragments/proxmox.py`.

Will work on that and submit again :)

File: changelogs/fragments/8440-allow-api-port-specification.yaml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - proxmox - allow specification of the API Port when using proxmox_* (https://github.com/ansible-collections/community.general/issues/8440, https://github.com/ansible-collections/community.general/pull/8441).
```
Comment (felixfontein) [inline changelogs/fragments/8440-allow-api-port-specification.yaml:2]:
```suggestion
  - proxmox - allow specification of the API port when using proxmox_* (https://github.com/ansible-collections/community.general/issues/8440, https://github.com/ansible-collections/community.general/pull/8441).
```

File: plugins/doc_fragments/proxmox.py [hunk 4]
```diff
@@ -16,6 +16,11 @@ class ModuleDocFragment(object):
       - Specify the target host of the Proxmox VE cluster.
     type: str
     required: true
+  api_port:
+    description:
+      - Specify the target port of the Proxmox VE cluster.
+    type: int
+    required: false
```
Comment (felixfontein) [inline plugins/doc_fragments/proxmox.py:23]:
```suggestion
    required: false
    version_added: 9.1.0
```

File: plugins/doc_fragments/proxmox.py [hunk 5]
```diff
@@ -16,6 +16,11 @@ class ModuleDocFragment(object):
       - Specify the target host of the Proxmox VE cluster.
     type: str
     required: true
+  api_port:
+    description:
+      - Specify the target port of the Proxmox VE cluster.
```
Comment (felixfontein) [inline plugins/doc_fragments/proxmox.py:21]:
```suggestion
      - Specify the target port of the Proxmox VE cluster.
      - Uses the E(PROXMOX_PORT) environment variable if not specified.
```

**Comment by felixfontein** [comment]
If nobody objects, I'll merge this in ~a week.

**Comment by felixfontein** [comment]
@tgouverneur thanks for your contribution!

**Pull request merged**